src/main/java/org/treadmill/zk/plugin/matcher/RoleMatcher.java

```diff
@@ -0,0 +1,49 @@
+package org.treadmill.zk.plugin.matcher;
+
+import com.google.common.cache.CacheBuilder;
+import com.google.common.cache.CacheLoader;
+import com.google.common.cache.LoadingCache;
+import com.google.inject.Inject;
+import com.google.inject.Singleton;
+import com.unboundid.ldap.sdk.LDAPSearchException;
+import org.slf4j.Logger;
+import org.treadmill.zk.plugin.utils.LdapQuery;
+
+import java.io.IOException;
+import java.util.Set;
+import java.util.concurrent.ExecutionException;
+
+import static org.slf4j.LoggerFactory.getLogger;
+import static org.treadmill.zk.plugin.utils.Configuration.get;
+
+@Singleton
+public class RoleMatcher extends Matcher {
+
+  private static final Logger logger = getLogger(RoleMatcher.class);
+  private LoadingCache<String, Set<String>> cache;
+
+
+  @Inject
+  public RoleMatcher(LdapQuery ldapQuery) throws IOException {
+    setupCache(ldapQuery);
+  }
+
+  @Override
+  public boolean matchAcl(String principal, String role) throws ExecutionException {
+    if ("readers".equals(role)) return true;
```

Comment:
curios why do you need to hardcode readers at this level. readermatcher is gone? cause user can also be in a role (uid treadmld as "admin" role on zk) 

it seems like matcher has "host/" hardcoded but i think it should rather construct full principal when building the cache so that host/<...>@somerealm is in the cache together with someuser@somerealm.